Classes/Uploaders/CustomUploader.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json.Linq;
```

Comment:
I will have a look if this can be done using the system package without adding too much complexity. However if it results in a much higher complexity i would prefer not to use the system package.

Reply:
I did a little bit of research and saw that this JsonPath feature is currently not supported in `System.Text.Json` (https://github.com/dotnet/runtime/issues/31068).

So it you might be right in that it will result in much higher complexity. I think we may have to consider importing this package if we have no other choice.

Reply:
I've had a look at the system package to parse the json accordingly. It would require to implement a "custom" way of JsonPath by splitting the JsonPath string and navigation down the json by calling GetProperty on the nodes. 
Not sure if it was me who caused an error of if it just doesn't work as expected, but i could get the JsonDocument to navigate down the tree when the root node is selected first (which should be required, to make sure the very top of the tree is selected).

When having a Json object like:
```json
{
  "files": [
    "someUrl.de"
  ]
}
```
This didn't work for me:
```c#
JsonDocument doc = JsonDocument.Parse(jsonobject);
RootElement root = jsonParse.RootElement

// [...] Split jsonpath string and loop through it [...]
JsonElement filesNode = root.GetProperty("files");
```

Reply:
I took another look at this and agree that we would need to re-implement all of query options of JSONPath. I think it would be a bad idea for us to do that so I looked into this [JsonPath.Net](https://www.nuget.org/packages/JsonPath.Net) package which revolves around `System.Text.Json`

Here is some example code I put together:
```c#
using Json.Path;
using System.Text.Json.Nodes;

string json = @"
{
    ""firstName"": ""John"",
    ""lastName"": ""doe"",
    ""age"": 26,
    ""address"": {
        ""streetAddress"": ""naist street"",
        ""city"": ""Nara"",
        ""postalCode"": ""630-0192""
    },
    ""phoneNumbers"": [
        {
            ""type"": ""iPhone"",
            ""number"": ""0123-4567-8888""
        },
        {
            ""type"": ""home"",
            ""number"": ""0123-4567-8910""
        }
    ]
}";

if (JsonPath.TryParse("$.phoneNumbers[:1].type", out JsonPath path)) {
    var results = path.Evaluate(JsonNode.Parse(json));
    Console.WriteLine(results.Matches[0].Value); // outputs "iPhone"
}
```
I think this package is more lightweight compared against Newtonsoft's Json.NET, since it is acting as a utility package for `System.Text.Json`. I'd figure to use this for now and dropping the package once JsonPath is officially supported by Microsoft.

If this is okay, could you try this approach and see if it works for you? I can try implementing this, but I don't have a proper use case for this upload feature so I might not be able to properly test it.

Reply:
That approach sounds promising. I'll try to implement this at the weekend. Shouldn't be too hard.

I would like to support XML using Xpath (as described in the PR) but i cannot find anytime to propely test and implement it. Would you agree that XPath should be left out for now and be implemented later on demand?

Reply:
> Would you agree that XPath should be left out for now and be implemented later on demand?

Yep, that sounds fine to me.